Sources/SwiftlyCore/HTTPClient.swift

```diff
@@ -22,6 +22,101 @@ private func makeRequest(url: String) -> HTTPClientRequest {
     return request
 }
 
+struct SwiftOrgSwiftlyRelease: Codable {
+    var name: String
+}
+
+struct SwiftOrgPlatform: Codable {
+    var name: String
+    var archs: [String]
+
+    /// platformName is a mapping from the 'name' field of the swift.org platform object
+    /// to swiftly's PlatformDefinition name.
+    var platformName: String {
```

Comment:
To better align with this var's usage/documentation, it might be better to just return the matching `PlatformDefinition` or `nil` if it isn't supported. That way we can rely on the `PlatformDefinition` constants as the single source of supported platform names, and we also don't need to define constants for platforms we don't support. If we go that direction, renaming this something like `asPlatformDefinition` or something might make sense.

If we don't change this in that way, then I think at least renaming it to `canonicalPlatformName` or `swiftlyPlatformName` communicates the difference between the other `name` field a bit better. We'd also want to use the `PlatformDefinition` name constants here as well.

Reply:
Thanks, this is a really nice pattern here. We can return the constants defined in PlatformDefinition for platforms that swiftly supports, and manually constructed PlatformDefinitions for the ones that it doesn't, which can provide enough metadata to represent what the swift.org API is producing. For yet unknown (to swiftly) swift.org platforms, then it's just nil until this code adds a representation for it.